ouroboros/chainsync/typesV6.go

```diff
@@ -0,0 +1,155 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package chainsync
+
+import (
+	"encoding/json"
+	"strconv"
+
+	"github.com/SundaeSwap-finance/ogmigo/ouroboros/chainsync/num"
+)
+
+// TODO - Probably a better way to handle this. For now, just split up and handle
+// the differences manually. Also, we need to check the direction. Backward & forward
+// are different.
+type ResultPraosV6 struct {
+	Direction string   `json:"direction,omitempty" dynamodbav:"direction,omitempty"`
+	Block     *BlockV6 `json:"block,omitempty"     dynamodbav:"block,omitempty"`
+	Tip       *TipV6   `json:"tip,omitempty"       dynamodbav:"tip,omitempty"`
+}
+
+type ResponsePraosV6 struct {
+	JsonRpc string          `json:"jsonrpc,omitempty" dynamodbav:"jsonrpc,omitempty"`
+	Method  string          `json:"method,omitempty"  dynamodbav:"method,omitempty"`
+	Result  *ResultPraosV6  `json:"result,omitempty"  dynamodbav:"result,omitempty"`
+	ID      json.RawMessage `json:"id,omitempty"      dynamodbav:"id,omitempty"`
+}
+
+type BlockV6 struct {
+	Type         string        `json:"type,omitempty"`
+	Era          string        `json:"era,omitempty"`
+	ID           string        `json:"id,omitempty"`
+	Ancestor     string        `json:"ancestor,omitempty"`
+	Nonce        NonceV6       `json:"nonce,omitempty"`
+	Height       uint64        `json:"height,omitempty"`
+	Size         BlockSizeV6   `json:"size,omitempty"`
+	Slot         uint64        `json:"slot,omitempty"`
+	Transactions []TxV6        `json:"transactions,omitempty"`
+	Protocol     ProtocolV6    `json:"protocol,omitempty"`
+	Issuer       BlockIssuerV6 `json:"issuer,omitempty"`
+}
+
+type BlockIssuerV6 struct {
+	VerificationKey        string        `json:"verificationKey,omitempty"`
+	VrfVerificationKey     string        `json:"vrfVerificationKey,omitempty"`
+	OperationalCertificate OpCertV6      `json:"operationalCertificate,omitempty"`
+	LeaderValue            LeaderValueV6 `json:"leaderValue,omitempty"`
+}
+
+type OpCertV6 struct {
+	Count uint64 `json:"count,omitempty"`
+	Kes   KesV6  `json:"kes,omitempty"`
+}
+
+type LeaderValueV6 struct {
+	Proof  string `json:"proof,omitempty"`
+	Output string `json:"output,omitempty"`
+}
+
+type KesV6 struct {
+	Period          uint64 `json:"period,omitempty"`
+	VerificationKey string `json:"verificationKey,omitempty"`
+}
+
+type ProtocolV6 struct {
+	Version ProtocolVersion `json:"version,omitempty" dynamodbav:"version,omitempty"`
+}
+
+type NonceV6 struct {
+	Output string `json:"output,omitempty" dynamodbav:"slot,omitempty"`
+	Proof  string `json:"proof,omitempty"  dynamodbav:"slot,omitempty"`
+}
+
+type BlockSizeV6 struct {
+	Bytes int64
+}
+
+type TxV6 struct {
+	ID                       string                `json:"id,omitempty"                       dynamodbav:"id,omitempty"`
+	Spends                   string                `json:"spends,omitempty"                   dynamodbav:"spends,omitempty"`
```

Comment:
what is the `Spends` field? :thinking:

Reply:
oh probably Inputs vs Collateral

Reply:
Yeah, the only allowed values are "inputs" and "collaterals". (Actually, several of these fields have only one or two allowable values. Should we try to enforce that at the library level? I want to say yes but I don't want to make the code too overbearing with checks.)

Reply:
no, no use in that IMO; it's the kind of thing that feels good to do, but doesn't provide much value here, mainly because the user isn't producing these values, something like ogmios is